### ISSUE TITLE
A false negative about the rule ES_COMPARING_STRINGS_WITH_EQ

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#1764]
**SpotBugs Version:** 4.4.2
**Rule:** [ES_COMPARING_STRINGS_WITH_EQ](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq)

Hi! I find a false negative about this rule. SpotBugs should have reported a warning at line 4.

**Minimal Code Sample**

```java
public boolean func() {
    String s1 = "str1";
    final String s2 = "str2";
    return s1 == s2; // should report a warning here
}
```

**Expected outcome:**
SpotBugs should report a violation at line 4, but doesn't. This is a false-negative.

**Running SpotBugs through:** [CLI]
